### ISSUE TITLE
String printing on carrays with shape (n, 1)

### ISSUE THREAD

**Issue by mrocklin** [Blosc/bcolz#165]
These work fine:

```python
In [1]: import bcolz

In [2]: bcolz.ones((5, 2))
Out[2]: 
carray((5, 2), float64)
  nbytes: 80; cbytes: 16.00 KB; ratio: 0.00
  cparams := cparams(clevel=5, shuffle=True, cname='blosclz')
[[ 1.  1.]
 [ 1.  1.]
 [ 1.  1.]
 [ 1.  1.]
 [ 1.  1.]]

In [3]: bcolz.ones((0, 1))
Out[3]: 
carray((0, 1), float64)
  nbytes: 0; cbytes: 16.00 KB; ratio: 0.00
  cparams := cparams(clevel=5, shuffle=True, cname='blosclz')
[]

```

This fails

```python
In [4]: bcolz.ones((5, 1))
TypeError: object of type 'float' has no len()
```

I believe that this is because of the following

```python
In [5]: bcolz.ones((5, 1))[0]
Out[5]: 1.0
```

That should be an array, not a scalar

**Comment by esc** [comment]
Please submit a pull-request which exposes the issue with a test, has the suggested fix and lastly an entry to the changelog.

**Comment by FrancescAlted** [comment]
Fixed in #296 .
